ocs_ci/deployment/cnv.py

```diff
@@ -792,3 +797,92 @@ def uninstall_cnv(self, check_cnv_installed=True):
+
+    def cnv_patch_subscription(self, patch):
+        patch_cmd = (
+            f"oc -n {self.namespace} patch sub {constants.KUBEVIRT_HYPERCONVERGED} "
+            f"-p {patch} --type merge"
+        )
+        run_cmd(patch_cmd)
+
+    def get_running_cnv_version(self):
+        """
+        Get the currently deployed cnv version
+
+        Returns:
+            string: cnv version
+
+        """
+        occmd = (
+            f"oc get sub {constants.KUBEVIRT_HYPERCONVERGED} -n openshift-cnv -o json"
+        )
+        jq_cmd = "jq -r .status.currentCSV"
+        json_out = subprocess.Popen(shlex.split(occmd), stdout=subprocess.PIPE)
+        cnv_version = subprocess.Popen(
+            shlex.split(jq_cmd), stdin=json_out.stdout, stdout=subprocess.PIPE
+        )
+        json_out.stdout.close()
+        return cnv_version.communicate()[0].decode()
+
+    def check_cnv_is_upgradable(self):
+        """
+        This method checks if the cnv operator is upgradable or not
+        """
+        logger.info("Check cnv is installed")
+        assert self.cnv_hyperconverged_installed(), "cnv operator is not installed"
+
+        logger.info("Check cnv is READY")
+        assert self.post_install_verification(
+            raise_exception=False
+        ), "CNV operator is not ready"
+
+        cmd = (
+            f"oc get hyperconverged {constants.KUBEVIRT_HYPERCONVERGED} -n openshift-cnv -o json | "
+            "jq '.status.conditions'"
+        )
+        cmd_res = exec_cmd(cmd, shell=True)
+        if cmd_res.returncode != 0:
+            logger.error(f"Failed to disable multicluster engine\n{cmd_res.stderr}")
+            return
+        output = cmd_res.stdout.decode("utf-8")
+        logger.info(f"The output is: {output}")
+        data = json.loads(output)
+        upgradeable_present = any(item.get("type") == "Upgradeable" for item in data)
+        return upgradeable_present
+
+    def upgrade_cnv(self):
+        """
+        Upgrade cnv operator
+
+        """
+        if not self.check_cnv_is_upgradable():
+            logger.info("CNV is not upgradable")
+            return
+
+        logger.info("Currently installed cnv version")
+        print(
+            f" currently installed cnv version: {parse_version(self.get_running_cnv_version())}"
+        )
+        self.upgrade_version = config.UPGRADE.get("upgrade_cnv_version")
+        if not self.upgrade_version:
+            self.upgrade_version = get_running_ocp_version()
+        print(f"Upgarde cnv version: {parse_version(self.upgrade_version)}")
+
+        # we create catsrc with nightly builds only if config.DEPLOYMENT does not have cnv_latest_stable
+        if not config.DEPLOYMENT.get("cnv_latest_stable"):
+            # Create CNV catalog source
+            self.create_cnv_catalog_source()
+
+        # Update CNV subscription
+        patch = f'\'{{"spec": {{"channel": "nightly-{self.upgrade_version}"}}}}\''
+        self.cnv_patch_subscription(patch)
```

Comment:
Using class OCP or OCS makes the code maintainable.

```suggestion
        hyperconverged_subs = OCP(kind=constants.SUBSCRIPTION_WITH_ACM, namespace=self.namespace, resource_name=constants.KUBEVIRT_HYPERCONVERGED)
        hyperconverged_subs.patch(params=patch, format_type="merge")
```

And remove the new function  cnv_patch_subscription OR
use class OCP the function cnv_patch_subscription

Reply:
Addressed